node/src/Pos/Wallet/Web/Util.hs

```diff
@@ -33,66 +33,65 @@ import           Pos.Wallet.Web.ClientTypes (AccountId (..), Addr, CId,
 
 import           Pos.Wallet.Web.Error       (WalletError (..))
 import           Pos.Wallet.Web.State       (AddressLookupMode(..), AddressInfo (..),
-                                             CurrentAndRemoved (..),
-                                             WebWalletModeDB, getAccountAddrMaps, getAccountIds,
+                                             CurrentAndRemoved (getCurrent, getRemoved),
+                                             WalletSnapshot,
+                                             getAccountAddrMaps, getAccountIds,
                                              getAccountWAddresses, getWalletMeta,
                                              getAccountMeta)
 
-getAccountMetaOrThrow :: (WebWalletModeDB ctx m, MonadThrow m) => AccountId -> m CAccountMeta
-getAccountMetaOrThrow accId = getAccountMeta accId >>= maybeThrow noAccount
+getAccountMetaOrThrow :: MonadThrow m => WalletSnapshot -> AccountId -> m CAccountMeta
+getAccountMetaOrThrow ws accId = maybeThrow noAccount (getAccountMeta ws accId)
```

Comment:
Aye. To make sure we are on the same page, do you mean to simply changing its type to something like:

```haskell
getAccountMetaOrThrow :: WalletSnapshot -> AccountId -> Maybe CAccountMeta
```

And then "offload" upstream the fact we can `MonadThrow` in an arbitrary monad `m` (i.e. throw within the bigger monadic context, if needed, so that the exception can bubble up to the servant error handlers) ?